packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1774,6 +1774,41 @@ describe('TransactionController', () => {
 
       expect(delayMessengerMock.call).not.toHaveBeenCalled();
     });
+
+    it.each([
+      [
+        providerErrors.userRejectedRequest(),
+        'catches error when user reject approval',
+      ],
+      [
+        new Error('TestError'),
+        'catches error no code property in error object',
+      ],
+      [undefined, 'catches error when error is undefined'],
+    ] as unknown[] | [unknown])('%s', async (errorToThrow) => {
+      jest.spyOn(console, 'error').mockImplementation();
+
+      (
+        delayMessengerMock.call as jest.MockedFunction<any>
+      ).mockImplementationOnce(() => {
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw errorToThrow;
+      });
+
+      const txParams = {
+        from: ACCOUNT_MOCK,
+        hash: '1337',
+        id: 'mocked',
+        chainId: toHex(5),
+        status: TransactionStatus.unapproved,
+      };
+      const controller = newController();
+      controller.state.transactions.push(txParams as any);
+
+      controller.initApprovals();
+
+      expect(delayMessengerMock.call).toHaveBeenCalledTimes(1);
```

Comment:
This method is unique in that it has asynchronous side effects but it's not asynchronous itself as we don't need to wait for the results.

Because of that, how are we asserting the error catch is working in this test as if it failed, would it not be in an asynchronous context anyway so our test wouldn't know about it?

Would a better assertion be to check that `console.error` is called with our expected message? Except for when it's a rejection error.

Reply:
Now it's asserting the message using the `console.error` spy. :pray: